File: core/src/main/java/heroes/journey/entities/items/Item.java
package heroes.journey.entities.items;

import static heroes.journey.registries.Registries.ItemManager;

import heroes.journey.entities.tagging.Attributes;
import heroes.journey.registries.Registrable;
import lombok.Getter;

@Getter
public class Item extends Registrable {

  private final ItemSubType subType;
  private final int weight;
  private final int value;
  private final Attributes attributes;

  public Item(String id, String name, ItemSubType subType, int weight, int value, Attributes attributes) {
    super(id, name);
    this.subType = subType;
    this.weight = weight;
    this.value = value;
    this.attributes = attributes;
  }

  public ItemType getType() {
    return subType.getParentType();
  }

  public Item register() {
    return ItemManager.register(this);
  }
}

File: core/src/main/java/heroes/journey/entities/items/Item.kt
package heroes.journey.entities.items

import heroes.journey.entities.tagging.Attributes
import heroes.journey.modlib.items.IItem
import heroes.journey.modlib.items.ItemSubType
import heroes.journey.modlib.items.ItemType
import heroes.journey.modlib.registries.Registrable
import heroes.journey.mods.Registries.ItemManager
import lombok.Getter

@Getter
open class Item(
    id: String,
    override val subType: ItemSubType,
    override val weight: Int,
    override val attributes: Attributes = Attributes()
) :
    Registrable(id), IItem {

    override fun register(): Item {
        return ItemManager.register(this)
    }

    val type: ItemType
        get() = subType.parentType
}
